src/commands/appSettings/SqlDatabaseConnectionPromptStep.ts

```diff
@@ -0,0 +1,93 @@
+import { localize, skipForNow } from '../../localize';
+import { SqlServerListStep } from '../createFunction/durableSteps/sql/SqlServerListStep';
+import { IConnectionPromptOptions } from './IConnectionPrompOptions';
+import { ISqlDatabaseConnectionWizardContext } from './ISqlDatabaseConnectionWizardContext';
+import { SqlDatabaseConnectionCustomPromptStep } from './SqlDatabaseConnectionCustomPromptStep';
+
+export class SqlDatabaseConnectionPromptStep<T extends ISqlDatabaseConnectionWizardContext> extends AzureWizardPromptStep<T> {
+    public constructor(private readonly _options?: IConnectionPromptOptions) {
+        super();
+    }
+
+    public async prompt(context: T): Promise<void> {
+        if (this._options?.preSelectedConnectionType) {
+            context.sqlDbConnectionType = this._options.preSelectedConnectionType;
+            context.telemetry.properties.sqlDbConnectionType = this._options.preSelectedConnectionType;
+            return;
+        }
+
+        const connectAzureDatabase: MessageItem = { title: localize('connectSqlDatabase', 'Connect Azure SQL Database') };
+        const connectNonAzureDatabase: MessageItem = { title: localize('connectSqlDatabase', 'Connect Non-Azure SQL Database') };
+        const useExistingConnectionButton: MessageItem = { title: localize('useExistingConnection', 'Use Existing Connection') };
+        const skipForNowButton: MessageItem = { title: skipForNow };
+
+        const message: string = localize('selectSqlDatabaseConnection', 'In order to proceed, you must connect a SQL database for internal use by the Azure Functions runtime.');
+
+        const buttons: MessageItem[] = [connectAzureDatabase, connectNonAzureDatabase];
+
+        if (!this._options?.suppressSkipForNow) {
+            buttons.push(skipForNowButton);
+        } else if (this._options?.suppressSkipForNow && !context.sqlDbConnectionForDeploy) {
+            // On debug, give user the option to run from an existing local connection string
+            const existingConnection: string | undefined = await getLocalConnectionString(context, ConnectionKey.SQL, context.projectPath);
+            if (!!existingConnection) {
+                buttons.push(useExistingConnectionButton);
+            }
+        }
+
+        const result: MessageItem = await context.ui.showWarningMessage(message, { modal: true }, ...buttons);
+        if (result === connectAzureDatabase) {
+            context.sqlDbConnectionType = ConnectionType.Azure;
+        } else if (result === connectNonAzureDatabase) {
+            context.sqlDbConnectionType = ConnectionType.NonAzure;
+        } else {
+            context.sqlDbConnectionType = ConnectionType.None;
+        }
+
+        context.telemetry.properties.sqlDbConnectionType = context.sqlDbConnectionType;
+    }
+
+    public shouldPrompt(context: T): boolean {
+        if (context.azureWebJobsStorageType) {
+            context.sqlDbConnectionType = context.azureWebJobsStorageType;
```

Comment:
Similar to #3390, I'm not a fan of `shouldPrompt()` having effects. I think a better approach is to have it go to `prompt()`, but allow `prompt()` to not actually prompt, and just execute the effects automatically.

Reply:
Opened an issue for us to revisit:  https://github.com/microsoft/vscode-azurefunctions/issues/3465